### DISCUSSION TITLE
Improve experience for multi-language websites that are not 1:1

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco-CMS#14228] [Features and ideas]
Hi!

I know that the language variants feature that was added some years ago facilitates a nice 1:1 translation experience for websites but sometimes we still want to use the old "different root nodes for different languages"-approach, like this:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/d05e1059-6d16-4a09-8d77-9c92a883e532)

We often use this when the different version of the website looks very different between the markets/languages and also for sites that have been migrated from earlier versions where the only option for 1:1 translations was to use Vorto.

In the scenario I describe here we have three languages (Swedish, Danish and English) and one root node per language with a unique URL/domain configured per language.

The problem that we face is that Umbraco wants to generate URLs for all configured cultures on all nodes:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/f64d44b2-d25b-4d79-9e87-d9b7b35ac550)

In the screenshot above the only URLs that would make sense for the content editor would be the Swedish URLs but I can't seem to understand/find a way to make it work like this without hacks with UrlProviders etc.

Maybe I'm missing something here, maybe or approach is "wrong", or maybe there is settings that I don't know of but I wanted to raise the question here and see if there is an existing solution to this OR if maybe we could make improvements to make this experience better.

**Comment by nul800sebastiaan** [comment]
The "old" way of doing multilingual setups hasn't changed so is still completely possible. From these screenshots I can't really tell what is going on exactly but I think you might have missed to set up "Culture and hostnames" on the root nodes?

**Reply by nul800sebastiaan** [reply]
> maybe or approach is "wrong" 

The approach is very very "right"! 👍

**Comment by enkelmedia** [comment]
@nul800sebastiaan Thanks the reply =D

Cultures and hostnames was configured in the screenshot above, all three as subdomains of local host.

https://localhost/sv
https://localhost/en
https://localhost/dk

I'm talking about the fact that there is URLs for all cultures even when this specific page (based on the configured hostnames) only have a Swedish version. What I'm saying is that there should be a way to disable cultures when they are not configured so that only the URLs for the Swedish site is shown in the info tab.

I've also notices issues when using different domains (not sub-directories) for different cultures.

In the example below we have three root notes, each only have one configured culture and hostname.

* Home [EN] configured for www.mysite.com (en-US)
* Home [SV] configured for www.mysite.se (sv-SE)
* Home [DK] configured for www.mysite.dk (da-DK)

Here I'm accessing the backoffice from www.mysite.com and the Swedish homepage "Hem [SV]" shows one URL that is www.mysite.com/hem-sv/ (notice that this is sub-directory of the english version of the site.

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/2587aaed-8575-4152-8300-4afb1d056788)

I can also request this URL in the browser and the swedish homepage will be shown for as a sub-url on the english site:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/9977e480-124b-410e-afb5-000323f65bc0)

Like I said, it might be something wrong with my setup but this is how "we used to do it" and on upgraded site this behavoir is kind of confusing.

Happy to provice a ZIP/Rar with the setup if you want to look closer. Note also that this is on Umbraco 11.3.

**Reply by nul800sebastiaan** [reply]
It's a faint memory from the v4 days, but something about this setting rings a bell: https://docs.umbraco.com/umbraco-cms/reference/configuration/globalsettings#hide-top-level-nodes-from-path

**Comment by enkelmedia** [comment]
Thanks @nul800sebastiaan, I was around during these days as well and I've tried that setting without success =D

With the `HideTopLevelNodeFromPath`-setting configured as true, the info-tab still looks like this:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/63879d59-350b-4f9f-906f-9f06830edfa9)

I did re-generate the database cache, memory cache and so on, notice the link to the "English version" of the page? This link renders like this:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/77ec451b-bb11-44da-a909-b23403ef4480)

As you see, the culture is sv-SE but the link on the info-tab indicated that it was the English version. And there is only a Swedish hostname configured on that node:

![image](https://github.com/umbraco/Umbraco-CMS/assets/1782524/a80b92c9-793b-4e4f-ade8-cfe5c49fd6cb)

Things are obviously broken here and I've spent time looking at the source of the core and trying to get this to work in a way that makes sense but without success. I think that this is confusing both for me as a developer and especially for editors. I wanted to start this discussion to raise the issue and try to figure out if there is a way forward, either with other defaults or using configuration options. 

In this scenario above I think that since no English hostname is configured there should not even be any indication of/link to an English version of this page as it does not vary by culture and no English hostname is configured.

If I continue to think out loud it feels like languages configured in the Settings section are assumed to be used everywhere but in this scenario only one language per language node makes sense. One could argue that the hostname configuration indicates this but maybe there needs to be some kind of setting similar to the hostname-setting that indicates which cultures that applies to the node and its descendants. 

I would be happy to contribute in any way if/when I know the direction in which to address this.

Cheers!
